test/python/test_pool.py

```diff
@@ -0,0 +1,164 @@
+
+
+@given(parsers.parse('a mayastor instance "{name}"'))
+def get_instance(mayastor_instance):
+    pass
+
+@given('a pool')
+def create_pool(mayastor_instance, mayastor_state):
+    mayastor_state['pool'] = mayastor_instance.ms.CreatePool(pb.CreatePoolRequest(name='p0', disks=['malloc:///disk0?size_mb=100']))
+
+
+@when('the user attempts to create a pool specifying a URI representing an aio disk')
+def create_pool_from_aio_disk(mayastor_instance, create_image_file, mayastor_state):
+    try:
+        mayastor_state['pool'] = mayastor_instance.ms.CreatePool(pb.CreatePoolRequest(name='p0', disks=[f"aio://{create_image_file}"]))
+    except Exception as error:
+        mayastor_state['error'] = error
+
+
+@when('the user attempts to create a pool specifying a disk with an invalid block size')
+def create_pool_from_disk_with_an_invalid_block_size(mayastor_instance, mayastor_state):
+    try:
+        mayastor_state['pool'] = mayastor_instance.ms.CreatePool(pb.CreatePoolRequest(name='p0', disks=['malloc:///disk0?size_mb=100&blk_size=1024']))
+    except Exception as error:
+        mayastor_state['error'] = error
+
+
+@when('the user attempts to create a pool specifying multiple disks')
+def create_pool_from_multiple_disks(mayastor_instance, mayastor_state):
+    try:
+        mayastor_state['pool'] = mayastor_instance.ms.CreatePool(pb.CreatePoolRequest(name='p0', disks=['malloc:///disk0?size_mb=100', 'malloc:///disk1?size_mb=100']))
+    except Exception as error:
+        mayastor_state['error'] = error
+
+
+@when('the user attempts to create a pool with the name of the pool that already exists')
+def create_pool_that_already_exists(mayastor_instance, mayastor_state):
+    try:
+        mayastor_instance.ms.CreatePool(pb.CreatePoolRequest(name='p0', disks=['malloc:///disk0?size_mb=100']))
+    except Exception as error:
+        mayastor_state['error'] = error
+
+
+@when('the user attempts to destroy a pool that does not exist')
+def destroy_pool_that_does_not_exist(mayastor_instance, mayastor_state):
+    try:
+        mayastor_instance.ms.DestroyPool(pb.DestroyPoolRequest(name='p0'))
+    except Exception as error:
+        mayastor_state['error'] = error
+
+
+@when('the user attempts to destroy the pool')
+def destroy_pool(mayastor_instance, mayastor_state):
+    try:
+        mayastor_instance.ms.DestroyPool(pb.DestroyPoolRequest(name=mayastor_state['pool'].name))
+        del mayastor_state['pool']
+    except Exception as error:
+        mayastor_state['error'] = error
+
+
+@when('the user lists the current pools')
+def list_pools(mayastor_instance, mayastor_state):
+    try:
+        mayastor_state['pool_list'] = mayastor_instance.ms.ListPools(pb.Null(), wait_for_ready=True).pools
+    except Exception as error:
+        mayastor_state['error'] = error
+
+
+@then('the pool creation should fail')
+def pool_creation_should_fail(find_pool, mayastor_state):
```

Comment:
why not call this when the test should fail rather than storing an error which might be clobbered?

Reply:
As I understand it, this is not really a normal function to be invoked directly, it is much like a fixture and is invoked appropriately via the pytest infrastructure - which ensures that the argument values (again fixtures) are only evaluated once and the results cached appropriately.